### PR TITLE
[Enhancement] Save the memory to visualization backends

### PR DESCRIPTION
**Pull request by enkilee**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

add memory in other log

## Modification

add memory in other log

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Review by HAOCHENYE — COMMENTED** [review]
`log_str` means the logs displayed in the terminal or logs dumped into `xxx.log`. As described in #916, we want to see `memory` information in the saved JSON file, Tensorboard or WandB, therefore we should add `memory` filed to tag, rather than extend the `log_str`

**Comment by enkilee** [comment]
> `log_str` means the logs displayed in the terminal or logs dumped into `xxx.log`. As described in #916, we want to see `memory` information in the saved JSON file, Tensorboard or WandB, therefore we should add `memory` filed to tag, rather than extend the `log_str`

yes，but i read the code and docs, i can't find which code to generate json or other file and format the display information.

**Comment by zhouzaida** [comment]
Hi, you can modify the following lines

https://github.com/open-mmlab/mmengine/blob/b1b1f53db2712904c04ab61356fe257eb8e9803a/mmengine/runner/log_processor.py#L199-L200

to

```python
        if is_cuda_available():
            max_memory = self._get_max_memory(runner)
            log_str += f'memory: {max_memory}  '
            tag['memory'] = max_memory
            
```

File: mmengine/runner/log_processor.py [hunk 1]
```diff
@@ -271,6 +273,12 @@ def get_log_after_epoch(self,
             log_items.append(f'{name}: {val}')
         log_str += '  '.join(log_items)
 
+        # If cuda is available, the max memory occupied should be calculated.
+        if is_cuda_available():
+            max_memory = self._get_max_memory(runner)
+            log_str += f'memory: {max_memory}  '
+            tag['memory'] = max_memory
+
```
Comment (zhouzaida) [inline mmengine/runner/log_processor.py:281]: Hi, `get_log_after_epoch` does not need to add memory to `tag` to append it to `log_str`.

File: mmengine/runner/log_processor.py [hunk 2]
```diff
@@ -270,7 +272,6 @@ def get_log_after_epoch(self,
                 val = f'\n{val}\n'
             log_items.append(f'{name}: {val}')
         log_str += '  '.join(log_items)
-
```
Comment (HAOCHENYE) [inline mmengine/runner/log_processor.py:273]: we should keep the empty line to make the code cleaner.

**Review by zhouzaida — APPROVED** [review]

**Pull request merged**